src/datanode/src/region_server.rs

```diff
@@ -548,6 +548,18 @@ impl RegionServerInner {
                 },
                 None => return error::RegionNotFoundSnafu { region_id }.fail(),
             },
+            RegionChange::Catchup => match current_region_status {
+                Some(status) => match status.clone() {
+                    RegionEngineWithStatus::Registering(_) => {
+                        return error::RegionNotReadySnafu { region_id }.fail()
+                    }
+                    RegionEngineWithStatus::Deregistering(_) => {
+                        return error::RegionNotFoundSnafu { region_id }.fail()
+                    }
+                    RegionEngineWithStatus::Ready(engine) => engine,
+                },
+                None => return error::RegionNotFoundSnafu { region_id }.fail(),
+            },
```

Comment:
`RegionChange::Catchup` and `RegionChange::None` are completely same, can we merge them?

Reply:
`RegionChange::None` includes query operations; we don't need to register logical regions after a query.

Reply:
Got you, updated